Change download link/version for Boost v 1.76.0 - Ubuntu

alloynetworks opened the pull request:
Update the download links and change the latest available Boost version from 1.71 to 1.76

alloynetworks committed:
Change download link/version for Boost v 1.76.0

Update the download links and change the latest available Boost version from 1.71 to 1.75

mDuo13 commented:
I believe @seelabs found [some minor incompatibilities with Boost 1.76](https://github.com/ripple/rippled/pull/3819) so we might want to suggest Boost 1.75 for now. And yes, this should be ported to all the build instructions.

(Interestingly: the link checker picked up on this from the macOS build instructions, where it's actual hyperlink and not just text, but I didn't realize until now that it applies to all the builds.)

alloynetworks commented:
Are the incompatibilities only in the tests? @seelabs ?

mDuo13 commented:
I checked with @seelabs and confirmed that Boost 1.76 is indeed not compatible with rippled 1.7.2 or earlier. So Boost 1.75.0 is the recommended version for building, though any version from 1.70.0–1.75.0 should work.

alloynetworks committed:
Update to change boost version to 1.75.0

As described in comments, the latest compatible version of Boost is 1.75.0 and not 1.76.0

mDuo13 commented on content/tutorials/manage-the-rippled-server/installation/build-run-rippled-ubuntu.md:
```diff
@@ -75,9 +75,9 @@ These instructions use Ubuntu's APT (Advanced Packaging Tool) to install the sof
           **Tip:** This example uses 4 processes to build in parallel. The best number of processes to use depends on how many CPU cores your hardware has available. You can use `cat /proc/cpuinfo` to get information about your hardware's processor.
 
-      6. Set the environment variable `BOOST_ROOT` to point to the new `boost_1_71_0` directory. It's best to put this environment variable in your `.profile`, or equivalent, file for your shell so it's automatically set when you log in. Add the following line to the file:
+      6. Set the environment variable `BOOST_ROOT` to point to the new `boost_1_76_0` directory. It's best to put this environment variable in your `.profile`, or equivalent, file for your shell so it's automatically set when you log in. Add the following line to the file:
```
```suggestion
      6. Set the environment variable `BOOST_ROOT` to point to the new `boost_1_75_0` directory. It's best to put this environment variable in your `.profile`, or equivalent, file for your shell so it's automatically set when you log in. Add the following line to the file:
```

mDuo13 commented on content/tutorials/manage-the-rippled-server/installation/build-run-rippled-ubuntu.md:
```diff
@@ -75,9 +75,9 @@ These instructions use Ubuntu's APT (Advanced Packaging Tool) to install the sof
+      6. Set the environment variable `BOOST_ROOT` to point to the new `boost_1_76_0` directory. It's best to put this environment variable in your `.profile`, or equivalent, file for your shell so it's automatically set when you log in. Add the following line to the file:
 
-              export BOOST_ROOT=/home/my_user/boost_1_71_0
+              export BOOST_ROOT=/home/my_user/boost_1_76_0
```
```suggestion
              export BOOST_ROOT=/home/my_user/boost_1_75_0
```

mDuo13 commented on content/tutorials/manage-the-rippled-server/installation/build-run-rippled-ubuntu.md:
```diff
@@ -49,21 +49,21 @@ These instructions use Ubuntu's APT (Advanced Packaging Tool) to install the sof
 5. Compile Boost.
 
-    Version 1.7.0 of `rippled` requires Boost version 1.70.0 or higher. Because Boost version 1.70.0 or higher isn't available in the Ubuntu 18.04 (or 16.04) software repositories, you must compile it yourself. The following examples use Boost 1.71.0, which was the newest version at the time of writing.
+    Version 1.7.0 and higher of `rippled` requires Boost version 1.70.0 or higher. Because Boost version 1.70.0 or higher isn't available in the Ubuntu 18.04 (or 16.04) software repositories, you must compile it yourself. The following examples use Boost 1.75.0, which was the newest version compatible with `rippled` at the time of writing.
```
```suggestion
    Version 1.7.0 of `rippled` requires the Boost library and is compatible with Boost versions 1.70.0 to 1.75.0. The Ubuntu 18.04 (or 16.04) software repositories don't have a compatible Boost version, so you must compile it yourself. The following examples use Boost 1.75.0.
```

mDuo13 reviewed (changes requested):
I noticed a couple places it still said 1.75.0. I also suggested a rewrite of one of the paragraphs that was getting a little hard to follow.

If you don't mind, I'd like to take it from here to apply the same changes to the other build instructions too.

alloynetworks commented:
> I noticed a couple places it still said 1.75.0. I also suggested a rewrite of one of the paragraphs that was getting a little hard to follow.
> 
> If you don't mind, I'd like to take it from here to apply the same changes to the other build instructions too.

I think that makes sense!

mDuo13 committed:
Ubuntu build: edits per review
